lib/utils.sh

```diff
@@ -319,95 +319,109 @@ is_service_enabled() {
+    if [ "$ANSWER" = "y" ]; then
+        debug "Kernel option $KERNEL_OPTION enabled"
+        FNRET=0
+    elif [ "$ANSWER" = "n" ]; then
+        debug "Kernel option $KERNEL_OPTION disabled"
+        FNRET=1
     else
-        if [ "$MODPROBE_FILTER" != "" ]; then
-            DEF_MODULE="$($SUDO_CMD modprobe -n -v "$MODULE_NAME" 2>/dev/null | grep -E "$MODPROBE_FILTER" | tail -1 | xargs)"
-        else
-            DEF_MODULE="$($SUDO_CMD modprobe -n -v "$MODULE_NAME" 2>/dev/null | tail -1 | xargs)"
-        fi
+        debug "Kernel option $KERNEL_OPTION not found"
+        FNRET=2 # Not found
+    fi
+}
+is_kernel_module_disabled() {
+    # check if a kernel module is disabled in the modprobe configuration
+    local MODULE_NAME="$1"
+    FNRET=1
 
-        if [ "$DEF_MODULE" == "install /bin/true" ] || [ "$DEF_MODULE" == "install /bin/false" ]; then
-            debug "$MODULE_NAME is disabled (blacklist with override)"
-            FNRET=1
-        elif [ "$DEF_MODULE" == "" ]; then
-            debug "$MODULE_NAME is disabled"
-            FNRET=1
-        else
-            debug "$MODULE_NAME is enabled"
-            FNRET=0
-        fi
+    local module_is_disabled=0
+    # is it blacklisted ?
+    if grep -qE "\s?+[^#]?blacklist\s+$MODULE_NAME\s?$" /etc/modprobe.d/*.conf; then
+        debug "$MODULE_NAME is blacklisted"
+        module_is_disabled=1
+    # maybe it is overriden ? check files in /etc/modprobe.d/ for "install xyz /bin/(true|false)"
+    elif grep -qE "\s?+[^#]?install\s+$MODULE_NAME\s+/bin/(true|false)\s?$" /etc/modprobe.d/*.conf; then
+        debug "$MODULE_NAME is disabled"
+        module_is_disabled=1
+    fi
+
+    if [ "$module_is_disabled" -eq 1 ]; then
+        debug "$MODULE_NAME is disabled in modprobe config"
+        FNRET=0
+    fi
+}
+
+is_kernel_module_available() {
+    # check if a kernel module is loadable, in a non monolithic kernel
 
-        if [ "$($SUDO_CMD lsmod | grep -E "$MODULE_NAME" 2>/dev/null)" != "" ]; then
-            debug "$MODULE_NAME is enabled"
+    local KERNEL_OPTION="$1"
+    FNRET=1
+
+    is_kernel_monolithic
+    if [ "$IS_MONOLITHIC_KERNEL" -eq 1 ]; then
+        info "your kernel is monolithic, no need to check for module availability"
+        return
+    fi
+
+    # look if a module is present as a loadable module in ANY available kernel, per CIS recommendation
+    # shellcheck disable=2013
+    for config_file in $($SUDO_CMD grep -l "^$KERNEL_OPTION=" /boot/config-*); do
+        module_config=$($SUDO_CMD grep "^$KERNEL_OPTION=" "$config_file" | cut -d= -f 2)
+        if [ "$module_config" == 'm' ]; then
+            debug "\"${KERNEL_OPTION}=m\" found in $config_file as module"
             FNRET=0
         fi
+    done
+}
+
+is_kernel_module_loaded() {
+    # check if a kernel module is actually loaded
+    local KERNEL_OPTION="$1"
+    local LOADED_MODULE_NAME="$2"
+    FNRET=1
+
+    is_kernel_monolithic
+    if [ "$IS_MONOLITHIC_KERNEL" -eq 1 ]; then
```

Comment:
note : invert code accordingly, it's just an example everything need to be aligned